### DISCUSSION TITLE
Data Storage

### DISCUSSION THREAD

**Discussion by HenryGeorgist** [HydrologicEngineeringCenter/HEC-FDA#97] [Q&A] [answered]
Currently the Datastorage is part of the View Model. In general data storage is a model concern. I would like to open a discussion about data storage being migrated into the model.

**Comment by Brennan1994** [comment]
the SQL lite database adds a lot of complexity I don't love. I'd be interested in alternatives. I don't feel good about learning and reimplementing database manager into the model. I'm not absolutely opposed, but It give me bad feels. 

@HenryGeorgist could you lay out and briefly describe some alternatives?

**Comment by HenryGeorgist** [comment]
no requirement says we shall use sqlite. I do not propose migration of the previous datamanager.

The data is roughly some groupings of paired data and paired data uncertain with sparse attributes. I propose we store it in something simple, xml or json for instance.

we can add an interface for data storage something like "public void Write(string writecontext)" where write context could represent a file path or an s3 key value pair. And "public void Read(string readcontext)" which we can change out our data model at any later date.

**Comment by rnugent3** [comment]
Is this something we'd need to do for product to be available for use in December?

**Reply by HenryGeorgist** [reply]
it depends on how we wish to proceed. we should talk it through.  If we want a product by December, yes, starting fresh with a new datastore would be much simpler than trying to migrate the current database out of the VM. This decision impacts migration to the existing UI, so it should be something we discuss sufficently.

**Reply by rnugent3** [reply]
Thanks, Will. I'll add this to the agenda for our chat later this week.

**Comment by rnugent3** [comment, marked as answer]
we have been moving more and more data storage responsibilities to the model. I don't expect to have all data storage responsibilities owned by the model for 2.0, but for releases thereafter we can pursue 100% of relevant data storage owned by the model.
